### PR TITLE
Adds Lawgivers to the Armory, to the HoS Office, and to the IAA Office

### PR DESCRIPTION
**Pull request by LordFowl**
As indicated by the title the purpose and intent of this PR is to situate within the three aforementioned locations (those locations being again for point of reiteration the armory, the Head of Security's Officer, and the Internal Affairs Office) lawgivers. There is one lawgiver in the armory, one lawgiver in the Head of Security's Office, and two lawgivers in the Internal Affairs Office, one for each Internal Affairs Officer. In each of these locations is also situated an instructional pamphlet which dictates useful commands to the user.

The lawgiver in the armory is a spare, the lawgiver in the Head of Security's Office is meant to be used when there is no Internal Affairs Officer on board, and the lawgivers in the Internal Affairs Office are meant for standardized usage within the course of a round on any shift level, from green to blue to red to delta.

In addition, laser rifles, laser carbines, energy rifles, energy carbines, ion rifles, tasers, stun batons, sec pistols, detective revolvers, combat shotguns, and detective pistols now possess a special firing pin. While this firing pin is installed the weapon will not fire if the user's ID has been blacklisted from weapons permissions, or the user has no ID.

These lawgivers are a bureaucratic aid, and indeed have special commands instead of the usual commands, they are:
Flash - loads the lawgiver with a chemical shell a la flash pistols.
Weapons Access - the lawgiver fires a beam that blacklists ID's from using weapons with the above mentioned firing pin.
Approval Scan - the lawgiver fires a beam that dictates the target's loyalty level.
Fine Target - The lawgiver deducts a number of credits from the target's account as dictated by the lawgiver, and stores them in a charge-card inside the lawgiver that can be removed by the user.
Megaphone - Amplifies the user's message.
Joke - comforts the user in these trying times.

image

- [ ] Testing

### CODE REVIEW

**Comment by TripH** [comment]
+1

**Comment by ParadoxSpace** [comment]
HOTSHOT

**Comment by Alberyk** [comment]
Some issues; I am sure that some of the weapons that are getting the blacklisted pins are not only used by security.

**Comment by shedya** [comment]
I don't see the blacklisted weapons as an issue because it's a _blacklist,_ not a whitelist. By default, everyone can use guns just fine, it's just that the IAAs are now capable of actually contributing to the round by stripping antags of their looted weapons, forcing them to spend TC on other weaponry or be more creative than just raiding the armoury.
I like this PR. Finally, something to do for IAAs.

File: code/modules/projectiles/guns/energy/special.dm [hunk 1]
```diff
@@ -33,6 +34,144 @@
...
+		playsound(src.loc, 'sound/machines/twobeep.ogg', 50, 1)
+		blacklist = 0
+		fine = 0
+
+	else if(findtext(msg,"megaphone"))
+		visible_message("<span class='warning'>\icon[src] [src.name] states: <FONT size=3>\"[msg]\"</FONT></span>")
+		playsound(src.loc, 'sound/machines/warning-buzzer.ogg', 100, 1)
+
+	else if(findtext(msg,"joke"))
+		var/jokes = pick("I was going to attend the clairvoyants meeting, but it was cancelled due to unforeseen events.", "Two cannibals are eating a clown. One cannibal turns to the other and asks, \'Does this taste funnysto you?\'", \
```
Comment (OneOneThreeEight) [inline code/modules/projectiles/guns/energy/special.dm]: typo > "funnysto"

**Comment by Alberyk** [comment]
> I don't see the blacklisted weapons as an issue because it's a _blacklist,_ not a whitelist. By default, everyone can use guns just fine, it's just that the IAAs are now capable of actually contributing to the round by stripping antags of their looted weapons, forcing them to spend TC on other weaponry or be more creative than just raiding the armoury.
> I like this PR. Finally, something to do for IAAs. Though I'd give the lawgivers loyalty pins, considering both IAA and HoS are loyalty-implanted.

Merchants, raiders and mercs can get some of those weapons, I don't think it would make sense for those to come with firing pins that can be disabled by the lawgiver.

**Comment by ParadoxSpace** [comment]
Is there anything really stopping IAAs/HoS from just spamming weapons denied on anyone that isn't Security?

**Pull request closed without merging**